stackslib/src/net/stackerdb/sync.rs

```diff
@@ -69,19 +69,53 @@ impl<NC: NeighborComms> StackerDBSync<NC> {
             next_chunk_push_priority: 0,
             expected_versions: vec![],
             downloaded_chunks: HashMap::new(),
-            replicas: vec![],
-            connected_replicas: vec![],
+            replicas: HashSet::new(),
+            connected_replicas: HashSet::new(),
             comms,
             stackerdbs,
             request_capacity: MAX_CHUNKS_IN_FLIGHT,
             max_neighbors: MAX_DB_NEIGHBORS,
             total_stored: 0,
             total_pushed: 0,
+            last_run_ts: 0,
         };
         dbsync.reset(None, config)?;
         Ok(dbsync)
     }
 
+    /// Calculate the new set of replicas to contact.
+    /// This is the same as the set that was connected on the last sync, plus any
+    /// config hints and discovered nodes from the DB.
+    fn find_new_replicas(
+        &self,
+        mut connected_replicas: HashSet<NeighborAddress>,
+        network: Option<&PeerNetwork>,
+        config: &StackerDBConfig,
+    ) -> Result<HashSet<NeighborAddress>, net_error> {
+        // keep all connected replicas, and replenish from config hints and the DB as needed
+        let mut peers = config.hint_replicas.clone();
+        if let Some(network) = network {
+            let extra_peers: Vec<_> = PeerDB::find_stacker_db_replicas(
+                network.peerdb_conn(),
+                network.get_local_peer().network_id,
+                &self.smart_contract_id,
+                self.max_neighbors,
+            )?
+            .into_iter()
+            .map(|neighbor| NeighborAddress::from_neighbor(&neighbor))
+            .collect();
+            peers.extend(extra_peers);
```

Comment:
`.extend()` can be called on the `map()` result directly, avoiding the need for `.collect()` (and the intermediate vec).

```suggestion
            .map(|neighbor| NeighborAddress::from_neighbor(&neighbor));
            peers.extend(extra_peers);
```